src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs

```diff
@@ -40,7 +40,7 @@ public class ExecutionManager : IExecutionManager
     /// <summary>
     /// Initializes a new instance of the <see cref="ExecutionManager"/> class.
     /// </summary>
-    public ExecutionManager(IRequestData requestData) : this(TestPlatformEventSource.Instance, requestData)
+    public ExecutionManager(IRequestData requestData!!) : this(TestPlatformEventSource.Instance, requestData)
```

Comment:
The protected ctor doesn't throw on null `requestData`, that seems weird to require non-null for public but allow null for ones inheriting.

Reply:
That's a nice catch, thanks. I will make the protected one require non-null as well but I will let the public one like this. I know it's not strictly necessary since the constructors are chained but I want to future proof in case we'll ever remove the chaining.